File: eco-core/core-plugin/src/main/java/com/willfp/ecobosses/bosses/tick/tickers/TargetTicker.java
package com.willfp.ecobosses.bosses.tick.tickers;

import com.willfp.ecobosses.bosses.EcoBoss;
import com.willfp.ecobosses.bosses.tick.BossTicker;
import org.bukkit.entity.Entity;
import org.bukkit.entity.LivingEntity;
import org.bukkit.entity.Mob;
import org.bukkit.entity.Player;
import org.jetbrains.annotations.NotNull;

public class TargetTicker implements BossTicker {
    @Override
    public void tick(@NotNull final EcoBoss boss,
                     @NotNull final LivingEntity entity,
                     final long tick) {
        if (tick % 10 == 0) {
            for (Entity nearbyEntity : entity.getNearbyEntities(10, 5, 10)) {
                if (nearbyEntity instanceof Player && entity instanceof Mob) {
                    ((Mob) entity).setTarget((Player) nearbyEntity);
                }
            }
        }
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/ecobosses/tick/TargetTicker.kt
package com.willfp.ecobosses.tick

import com.willfp.ecobosses.bosses.LivingEcoBoss

class TargetTicker : BossTicker {
    override fun tick(boss: LivingEcoBoss, tick: Int) {
        val entity = boss.entity

        if (tick % 10 != 0) {
            return
        }

        entity.target = boss.boss.targetMode.getTarget(boss) ?: return
    }
}
